config/merge/merge_test.go

```diff
@@ -667,14 +745,32 @@ func TestMerge(t *testing.T) {
 					},
 				},
 			},
+			transcript: Transcript{[]Mapping{
```

Comment:
Not new to your code but is there a reason that this test specifically is the only HTTP headers related test that isn't testing removing a header?

Reply:
It's testing header lists in appended file fragments, which should never be merged but should instead produce two separate fragments in the merged config.